### DISCUSSION TITLE
Following prime's tutorial

### DISCUSSION THREAD

**Discussion by Sherifx2** [VonHeikemen/lsp-zero.nvim#374] [General]
I was following prime's tutorial and I noticed the first thing in the readme is totally calling me out. I was kind of confused while reading through the readme, by keybindings is that the remap file in the config? Also how do I get a language server active in a current file? Does that mean that I have to set up a language server in the file structure of each project I do?

**Comment by VonHeikemen** [comment]
One thing you should know: Primeagen overrides (literally) everything lsp-zero is trying to do. The way he used lsp-zero makes no sense.

So my advice is that you make your lsp configuration without lsp-zero. I have fork of his config that shows how to do that: [after/plugin/lsp.lua](https://github.com/VonHeikemen/prime-init.lua/blob/master/after/plugin/lsp.lua)

If you want to use lsp-zero version 3 with prime's setup, in this post you will find the snippet with the updated code: [Primeagen's config from 2022](https://lsp-zero.netlify.app/v3.x/blog/theprimeagens-config-from-2022.html)

> by keybindings is that the remap file in the config?

Keybindings and keymaps are the same thing. They are keyboard shortcuts.

Primeagen does not use lsp-zero's keymaps. You can ignore the keymaps on lsp-zero's readme.

> Also how do I get a language server active in a current file?

You use [nvim-lspconfig](https://github.com/neovim/nvim-lspconfig). Basically, you install a language server, then you call the setup function of that language server in your own Neovim config. `lspconfig` will try its best to determine what language server should be active.
